feature/amazon/src/test/java/com/revenuecat/purchases/amazon/PriceExtractorTest.kt

```diff
@@ -1,422 +1,305 @@
+        val (currencyCode, priceAmountMicros) = "$1,000.00".extractPrice("US")
         assertThat(currencyCode).isEqualTo("USD")
         assertThat(priceAmountMicros).isEqualTo(1_000_000_000)
     }
 
     @Test
     fun `US marketplace 7 USD US$`() {
-        mockCurrency("US$", "USD")
-
-        val (currencyCode, priceAmountMicros) = "US$7.00".extractPrice(mockCurrency, numberFormatWithDots())
+        val (currencyCode, priceAmountMicros) = "US$7.00".extractPrice("US")
         assertThat(currencyCode).isEqualTo("USD")
         assertThat(priceAmountMicros).isEqualTo(7_000_000)
     }
 
     @Test
     fun `US marketplace 7 USD US$ space`() {
-        mockCurrency("US$", "USD")
-
-        val (currencyCode, priceAmountMicros) = "US$ 7.00".extractPrice(mockCurrency, numberFormatWithDots())
+        val (currencyCode, priceAmountMicros) = "US$ 7.00".extractPrice("US")
         assertThat(currencyCode).isEqualTo("USD")
         assertThat(priceAmountMicros).isEqualTo(7_000_000)
     }
 
     @Test
     fun `US marketplace 7 USD US$ comma`() {
-        mockCurrency("US$", "USD")
-
-        val (currencyCode, priceAmountMicros) = "US$7,00".extractPrice(mockCurrency, numberFormatWithCommas())
+        val (currencyCode, priceAmountMicros) = "US$7,00".extractPrice("US")
         assertThat(currencyCode).isEqualTo("USD")
         assertThat(priceAmountMicros).isEqualTo(7_000_000)
     }
 
     @Test
     fun `US marketplace 7 USD US$ comma space`() {
-        mockCurrency("US$", "USD")
-
-        val (currencyCode, priceAmountMicros) = "US$ 7,00".extractPrice(mockCurrency, numberFormatWithCommas())
+        val (currencyCode, priceAmountMicros) = "US$ 7,00".extractPrice("US")
         assertThat(currencyCode).isEqualTo("USD")
         assertThat(priceAmountMicros).isEqualTo(7_000_000)
     }
 
     @Test
     fun `US marketplace 7 USD symbol after`() {
-        mockCurrency("$", "USD")
-
-        val (currencyCode, priceAmountMicros) = "7.00$".extractPrice(mockCurrency, numberFormatWithDots())
+        val (currencyCode, priceAmountMicros) = "7.00$".extractPrice("US")
         assertThat(currencyCode).isEqualTo("USD")
         assertThat(priceAmountMicros).isEqualTo(7_000_000)
     }
 
     @Test
     fun `US marketplace 7 USD comma symbol after`() {
-        mockCurrency("$", "USD")
-
-        val (currencyCode, priceAmountMicros) = "7,00$".extractPrice(mockCurrency, numberFormatWithCommas())
+        val (currencyCode, priceAmountMicros) = "7,00$".extractPrice("US")
         assertThat(currencyCode).isEqualTo("USD")
         assertThat(priceAmountMicros).isEqualTo(7_000_000)
     }
 
     @Test
     fun `US marketplace 7 USD US$ symbol after`() {
-        mockCurrency("US$", "USD")
-
-        val (currencyCode, priceAmountMicros) = "7.00US$".extractPrice(mockCurrency, numberFormatWithDots())
+        val (currencyCode, priceAmountMicros) = "7.00US$".extractPrice("US")
```

Comment:
i can deduce it from the tests, but might be nice to have in a comment somewhere of all the different formats we've encountered

Reply:
Some of these are made up just to make sure we are ready for the unexpected haha